linter/src/spawning/spawning.cpp

```diff
@@ -1,35 +1,94 @@
 #include "spawning.hpp"
 
+#include "config.h"
+
+#include <fmt/core.h>
+#include <glaze/json/read.hpp>
+
+#include <iostream>
+
+namespace bp = boost::process;
+
 namespace nutc {
 namespace spawning {
 
-void
-spawn_client(const std::string& uid, std::string& algoid)
+lint_result
+SpawnerManager::spawn_client(const std::string& algo_code)
 {
-    // we don't want to modify the original algo id because we want to use it later on
-    std::string sanitized_algoid = algoid;
-    std::replace(sanitized_algoid.begin(), sanitized_algoid.end(), '-', ' ');
-    pid_t pid = fork();
-
-    if (pid == 0) {
-        std::vector<std::string> args = {
-            "NUTC-linter-spawner", "--uid", uid, "--algoid", sanitized_algoid
-        };
-
-        std::vector<char*> c_args;
-        for (auto& arg : args)
-            c_args.push_back(arg.data());
-        c_args.push_back(nullptr);
-
-        execvp(c_args[0], c_args.data());
-
-        log_e(linting, "Failed to lint algoid {} for uid {}", sanitized_algoid, uid);
-
-        exit(1);
-    }
-    else if (pid < 0) {
-        log_e(linting, "Failed to fork");
-    }
+    auto in_pipe = std::make_shared<bp::async_pipe>(io_service_);
+    bp::opstream out_pipe;
+
+    auto child = std::make_shared<bp::child>(
+        bp::search_path("NUTC-linter-spawner"),
+        bp::std_in<out_pipe, bp::std_out> * in_pipe,
+        io_service_
+    );
+
+    out_pipe << algo_code << std::flush;
+    out_pipe.pipe().close();
+
+    auto kill_timer = std::make_shared<ba::steady_timer>(io_service_);
+    kill_timer->expires_after(ba::chrono::seconds(LINT_AUTO_TIMEOUT_SECONDS));
+
+    lint_result res;
+
+    kill_timer->async_wait([child, in_pipe, &res](const auto& ec) {
+        if (!ec) {
```

Comment:
```suggestion
        if (!ec) return;
```

you have the condition inverted.

![image](https://github.com/northwesternfintech/NUTC/assets/76668284/9b5db152-e516-4abe-b740-d3bb6f3737e0)

Reply:
<img width="572" alt="image" src="https://github.com/northwesternfintech/NUTC/assets/33611513/ec5c19c0-abdd-4761-8ee0-7aa0ea5013d4">
I believe its correct